crates/iceberg/src/expr/visitors/expression_evaluator.rs

```diff
@@ -223,11 +217,10 @@ impl BoundPredicateVisitor for ExpressionEvaluatorVisitor<'_> {
         literal: &Datum,
         _predicate: &BoundPredicate,
     ) -> Result<bool> {
-        let datum = reference.accessor().get(self.partition)?;
-
-        if Self::is_null(datum.literal()) {
-            return Ok(false);
-        }
+        let datum = match reference.accessor().get(self.partition)? {
```

Comment:
I think this is slightly neater with `let .. else`, but I'm not gonna lose any sleep over it 😅 

```rust
let Some(datum) = reference.accessor().get(self.partition)? else { return Ok(false); };
```

Reply:
I changed it - just in case ... can't be responsible for anyone missing their beauty sleep.